### PR TITLE
CSP-734 GetCalendarEvents region filters added code and coverage

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeAan.Web/Models/NetworkEventsViewModel.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@ public class NetworkEventsViewModel
     // model that builds the filter choices into a model that can populate FE
     public List<SelectedFilter> SelectedFilters { get; set; } = new List<SelectedFilter>();
 
+
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Models/NetworkEventsViewModel.cs:20]: 👾

**Review by DevenKShah — APPROVED** [review]

File: src/SFA.DAS.ApprenticeAan.Web/Models/NetworkEventsViewModel.cs (hunk 1 above)
Comment (MarkFCain) [reply src/SFA.DAS.ApprenticeAan.Web/Models/NetworkEventsViewModel.cs:18]: removed

File: src/SFA.DAS.ApprenticeAan.Web/Services/EventSearchQueryStringBuilder.cs [hunk 2]
```diff
@@ -58,18 +63,46 @@ public List<SelectedFilter> BuildEventSearchFilters(EventFilterChoices eventFilt
...
         var eventTypesChecklistFiltered = new List<ChecklistLookup>();
-        foreach (var a in eventFilterChoices.EventTypesLookup)
+        foreach (var a in eventFilterChoices.EventTypeChecklistDetails.Lookups)
         {
             eventTypesChecklistFiltered.AddRange(from b in eventFilterChoices.CalendarIds
                                                  where a.Value == b.ToString()
                                                  select a);
         }
 
+        foreach (var lookup in from lookup in eventFilterChoices.EventTypeChecklistDetails.Lookups from ef in eventFilterChoices.CalendarIds!.Where(ef => ef.ToString() == lookup.Value) select lookup)
```
Comment (DevenKShah) [inline src/SFA.DAS.ApprenticeAan.Web/Services/EventSearchQueryStringBuilder.cs:73]: 🤯

**Review by YuvrajPatil-DFE — COMMENTED** [review]
Some alignment issue in  SFA.DAS.ApprenticeAan.Domain/Interfaces/IOuterApiClient.cs

File: src/SFA.DAS.ApprenticeAan.Web/Views/NetworkEvents/_FilterDetails.cshtml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        <legend class="govuk-fieldset__legend govuk-fieldset__legend--m">
+            @Model.Title
+        </legend>
+        <div class="govuk-checkboxes govuk-checkboxes--small">
+            @foreach (var lookup in Model.Lookups)
+            {
+                <div class="govuk-checkboxes__item">
+                    <input class="govuk-checkboxes__input" id="type-employer" name="@Model.QueryStringParameterName" type="checkbox" value="@lookup.Value"
+                       @lookup.Checked >
+                    <label class="govuk-label govuk-checkboxes__label" for="type-employer">
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/SFA.DAS.ApprenticeAan.Web/Views/NetworkEvents/_FilterDetails.cshtml:15]: Good spot! Fixed.

**Review by YuvrajPatil-DFE — APPROVED** [review]

**Pull request merged**